torch/csrc/jit/script/compiler.cpp

```diff
@@ -470,6 +485,73 @@ struct to_ir {
     }
   }
 
+  void emitFor(const For& stmt) {
+    // For now, we only support range loops. e.g. for i in range(3): ...
+
+    auto targets = stmt.targets();
+    auto itrs = stmt.itrs();
+    auto body = stmt.body();
+
+    // itrs must consist of a single Apply node
+    if (stmt.itrs().size() != 1) {
+      throw ErrorReport(stmt)
+          << "List of iterables is not supported currently.";
+    }
+    if (itrs[0].kind() != TK_APPLY) {
+      throw ErrorReport(stmt)
+          << "Non-range for loops are currently not supported.";
+    }
+
+    Apply range_iterator = Apply(itrs[0]);
+    if (range_iterator.callee().kind() != TK_VAR) {
+      throw ErrorReport(stmt)
+          << "Non-range for loops are currently not supported.";
+    }
+
+    {
+      Var var = Var(range_iterator.callee());
+      if (var.name().name() != "range") {
+        throw ErrorReport(stmt)
+            << "Non-range for loops are currently not supported.";
+      }
+    }
+
+    List<Expr> args = range_iterator.inputs();
+    // TODO: start, stop, step loop
+    if (args.size() != 1) {
+      throw ErrorReport(stmt)
+          << "range() expects one argument but got" << args.size();
+    }
+
+    auto target_list = List<Ident>(targets);
+    std::vector<std::string> target_idents;
+    target_idents.reserve(target_list.size());
+    for (const auto& t : target_list) {
+      target_idents.push_back(Ident(t).name());
+    }
+
+    Value* max_trip_count = emitExpr(args[0], 1)[0];
+    Value* cond_value_dummy = emitBooleanConst(stmt, true)[0];
+
+    emitLoopCommon(
+        max_trip_count,
+        cond_value_dummy,
+        stmt.body(),
+        stmt,
+        nullptr,
```

Comment:
Can be replaced with `CompoundTree::create(TK_TRUE)`

Reply:
Where?